### PR TITLE
feat(chat): support attachment of folders (Issue #967)

### PR DESCRIPTION
**Pull request by IlyaBondar**
**Description:**

support attachment of folders

Issues:

- Issue #967 

**UI changes**

image

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1363.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1363.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1363.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/ChatMessage/ChatMessageContent.tsx [hunk 1]
```diff
@@ -230,6 +242,7 @@ export const ChatMessageContent = ({
 
     const attachments = getUserCustomContent(
       newEditableAttachments,
+      [], // TODO
```
Comment (denys-kolomiitsev) [inline apps/chat/src/components/Chat/ChatMessage/ChatMessageContent.tsx]: What is this **TODO** about? 
I think it needs to be more info.
Or removed if it is done already.

File: apps/chat/src/utils/app/file.ts [hunk 2]
```diff
@@ -370,3 +381,6 @@ export const getNextFileName = (
 
 export const prepareFileName = (filename: string) =>
   `${getFileNameWithoutExtension(filename)}${getFileNameExtension(filename)}`;
+
+export const FOLDER_ATTACHMENT_CONTENT_TYPE =
```
Comment (denys-kolomiitsev) [inline apps/chat/src/utils/app/file.ts]: What do you think to move this constant to the **constants**.

**Comment by IlyaBondar** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/475170)

File: apps/chat/src/components/Chat/ChatMessage/ChatMessageContent.tsx [hunk 3]
```diff
@@ -409,7 +431,19 @@ export const ChatMessageContent = ({
...
+                        )}
+                        folders={
+                          canAttachFolders
+                            ? (newEditableAttachments.filter(
+                                (f) =>
+                                  f.contentType ===
+                                  FOLDER_ATTACHMENT_CONTENT_TYPE,
+                              ) as unknown as FileFolderInterface[])
+                            : undefined
+                        }
```
Comment (denys-kolomiitsev) [inline apps/chat/src/components/Chat/ChatMessage/ChatMessageContent.tsx]: Maybe for the readability it is better to extract all this lines into the variable?
What do you think?

File: apps/chat/src/components/Chat/ChatMessage/ChatMessageContent.tsx [hunk 4]
```diff
@@ -440,7 +474,14 @@ export const ChatMessageContent = ({
...
                       }
-                      selectedFilesIds={newEditableAttachmentsIds}
+                      selectedFilesIds={newEditableAttachments.map((f) =>
+                        f.contentType === FOLDER_ATTACHMENT_CONTENT_TYPE
+                          ? ApiUtils.decodeApiUrl(f.id).replace(
+                              new RegExp('^metadata/'),
+                              '',
+                            ) + '/'
+                          : f.id,
+                      )}
```
Comment (denys-kolomiitsev) [inline apps/chat/src/components/Chat/ChatMessage/ChatMessageContent.tsx]: Maybe for the readability it is better to extract all this lines into the variable?

File: apps/chat/src/components/Files/AttachButton.tsx [hunk 5]
```diff
@@ -111,7 +117,16 @@ export const AttachButton = ({
     ],
   );
 
-  if (!canAttachFiles && !canAttachLinks) return null;
+  if (!canAttachFiles && !canAttachFolders && !canAttachLinks) return null;
+
+  // eslint-disable-next-line no-console
+  console.log('AttachButton - selectedFilesIds:', selectedFilesIds);
```
Comment (denys-kolomiitsev) [inline apps/chat/src/components/Files/AttachButton.tsx]: Do you need this `console.log`  here?

File: apps/chat/src/components/Files/FileManagerModal.tsx [hunk 6]
```diff
@@ -269,15 +387,29 @@ export const FileManagerModal = ({
...
-      setSelectedFilesIds([]);
+    if (deletingFileIds.length) {
+      dispatch(FilesActions.deleteFilesList({ fileIds: deletingFileIds }));
+      if (selectedFilesIds === deletingFileIds) {
+        setSelectedFilesIds([]);
+      }
+    }
+    if (deletingFolderIds.length) {
+      // TODO: implement
+      // dispatch(FilesActions.deleteFolderList({ folderIds: deletingFolderIds }));
```
Comment (denys-kolomiitsev) [inline apps/chat/src/components/Files/FileManagerModal.tsx:401]: Will it be implemented later?

Comment (IlyaBondar) [reply apps/chat/src/components/Files/FileManagerModal.tsx:401]: yes, not in current sprint

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1363.nightly-test.deltixhub.io
>E2E tests status: success

**Review by denys-kolomiitsev — APPROVED** [review]

**Pull request merged**